src/tree/updater_quantile_hist.cc

```diff
@@ -633,24 +626,34 @@ void QuantileHistMaker::Builder<GradientSumT>::Update(
   builder_monitor_.Stop("Update");
 }
 
+template <size_t kUnrollLen = 8>
+struct SparsePageView {
```

Comment:
So we can't use the view directly?

Reply:
Right now we can't. It is declared in `cpu_predictor.cc` and the same class (with different behaviour) is used for gpu (https://github.com/dmlc/xgboost/blob/master/src/predictor/gpu_predictor.cu#L30), so we can't just move it to the header. But we can probably rename both classes and add them to one of `common/..` files. If you think it will look good, please suggest a place for them.

Reply:
Sorry, I misunderstood you. Yes, this class doesn't seem to be doing anything useful :)
Removed it